### PR TITLE
Initial RBAC support: create and use Kubernetes service account for E…

### PR DESCRIPTION
**Pull request by maartenvandenbogaard**
…S cluster

ES cluster runs with a dedicated service account instead of default.
Can be handy in case one wants to bind the service account
to a pod security policy (PSP) role binding.

### CODE REVIEW

**Comment by wmuizelaar** [comment]
This solves a part of https://github.com/kubedb/project/issues/390

File: pkg/controller/rbac.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	core_util "github.com/appscode/kutil/core/v1"
+	api "github.com/kubedb/apimachinery/apis/kubedb/v1alpha1"
+	core "k8s.io/api/core/v1"
+	kerr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/reference"
+)
+
+func (c *Controller) deleteServiceAccount(elasticsearch *api.Elasticsearch) error {
```
Comment (the-redback) [inline pkg/controller/rbac.go]: This function is not used from anywhere at this moment. The deletion is handled by ensuring `ownerreference` while creation of `ServiceAccount`. https://github.com/kubedb/elasticsearch/pull/245/files#diff-70a3bbc3c09da335262060f7399ea499R36 

So, we can delete this method.

File: pkg/controller/rbac.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		},
+		func(in *core.ServiceAccount) *core.ServiceAccount {
+			core_util.EnsureOwnerReference(&in.ObjectMeta, ref)
+			return in
+		},
+	)
+	return err
+}
+
+func (c *Controller) deleteRoleBinding(elasticsearch *api.Elasticsearch) error {
```
Comment (the-redback) [inline pkg/controller/rbac.go]: This method is not used either. We can delete this too.

File: pkg/controller/rbac.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	if err := c.createServiceAccount(elasticsearch); err != nil {
+		if !kerr.IsAlreadyExists(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) deleteRBACStuff(elasticsearch *api.Elasticsearch) error {
```
Comment (the-redback) [inline pkg/controller/rbac.go]: Same goes here. We can delete this too.

**Comment by the-redback** [comment]
Thanks, @maartenvandenbogaard for the PR. :100: 

I have requested some changes. May be we can Fix these in later commit, as it is there in already RBAC supported repo.

**Pull request merged**